### PR TITLE
Add the whereami library source code and license to third-party

### PR DESCRIPTION
**Pull request by daviditen**
Add the whereami library for use in portably determining where a running
binary is located within the filesystem.

Update third-party/README and LICENSE to include whereami.

### CODE REVIEW

**Comment by gbtitus** [comment]
@daviditen Would you make a follow-up issue to say that our `make install` infrastructure needs to be updated to include the `third-party/whereami` stuff, please?  Don't assign it to anyone; I don't think you need to do the work but I'm not sure who does.  I'll handle figuring that out.  (Might be RAD/BTR.)

**Comment by daviditen** [comment]
Thanks! I opened #15884 .

**Comment by bradcray** [comment]
I'm not finding it quickly, but this should close an issue that was reported by a user, shouldn't it?

File: LICENSE [hunk 1]
```diff
@@ -79,6 +80,7 @@ The Chapel implementation is composed of two categories of code:
      re2              only used when CHPL_REGEXP is 're2'
      utf8-decoder     bundled into the Chapel runtime to decode UTF-8 strings
      libunwind        only used when CHPL_UNWIND is 'libunwind'
+     whereami         used when starting a program with a launcher
```
Comment (bradcray) [inline LICENSE:83]: @gbtitus: Reviewing this makes me realize belatedly:  We should have probably added an entry to this second "when is it used?" table for libfabric?

**Comment by daviditen** [comment]
> I'm not finding it quickly, but this should close an issue that was reported by a user, shouldn't it?

The issue is #15489 . My next PR #15885 is the one that should close it.

File: LICENSE (hunk 1 above)
Comment (gbtitus) [reply LICENSE:83]: Oops, yeah, I'll fix that.  Thanks @bradcray!

**Pull request merged**